From 938373ca38576c29fc4479c73a395cd5553a160b Mon Sep 17 00:00:00 2001
From: "Wu, Qiming" <qi-ming.wu@intel.com>
Date: Fri, 6 Jul 2018 05:49:27 +0200
Subject: [PATCH] Merge pull request #48 in SW_UGW/linux from
 bugfix/DRVLIB_SW-772-PMAC-control-register-disable-LCHKS to master

* commit '2ee36396406c483cb64fcb0c2830b50b95ea65a1':
  DRVLIB_SW-772: disable LCHKS in pmac control 2
---
 drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c b/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c
index 062e71d7c180..1e027afbdaca 100644
--- a/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c
+++ b/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c
@@ -807,6 +807,9 @@ static int ltq_gsw_pmac_init(void)
 	/* Enable the CPU port MAC address spoofing detection */
 	gsw_reg_set_bit(ops, 0x480, 0x4000);
 
+	/*PMAC control register 2 to disable LCHKS*/
+	gsw_reg_clr_bit(ops, 0xd05, 0x3);
+
 	/* Do the GSW-R configuration */
 	ops = gsw_get_swcore_ops(1);
 
@@ -957,6 +960,9 @@ static int ltq_gsw_pmac_init(void)
 
 	gsw_reg_set_val(ops, 0x455, 0x1);
 
+	/*PMAC control register 2 to disable LCHKS*/
+	gsw_reg_clr_bit(ops, 0xd05, 0x3);
+
 	pr_info("\n\t GSW PMAC Init Done!!!\n");
 	return 0;
 }
